src/server/scripts/Outland/CoilfangReservoir/TheUnderbog/boss_hungarfen.cpp

```diff
@@ -167,20 +168,26 @@ class spell_hungarfen_putrid_mushroom_primer : public SpellScript
         return ValidateSpellInfo({ SPELL_SUMMON_UNDERBOG_MUSHROOM });
     }
 
-    void HandleScript(SpellEffIndex /*effIndex*/)
+    /// @todo: Check if something else should be done here
+    void FilterTargets(std::list<WorldObject*>& targets)
     {
-        Creature* caster = GetCaster()->ToCreature();
-        if (!caster || !caster->IsAIEnabled())
+        if (targets.empty())
             return;
 
-        // This is wrong. They don't need a script to simply select random target. But what should be done here? FilterTargets? Exclude pets?
-        if (Unit* target = caster->AI()->SelectTarget(SelectTargetMethod::Random, 0))
-            target->CastSpell(target, SPELL_SUMMON_UNDERBOG_MUSHROOM, true);
+        WorldObject* target = Trinity::Containers::SelectRandomContainerElement(targets);
+        targets.clear();
+        targets.push_back(target);
+    }
+
+    void HandleScript(SpellEffIndex /*effIndex*/)
+    {
+        GetHitUnit()->CastSpell(GetHitUnit(), SPELL_SUMMON_UNDERBOG_MUSHROOM);
```

Comment:
no CastArgs at all, or what spell is triggering the spell, etc?

Reply:
triggered was neccessary, you don't want either the mushroom to fail because player is casting something or players cast to be interrupted